### DISCUSSION TITLE
Issues with shell-escape when building with texlab and tectonic

### DISCUSSION THREAD

**Discussion by JurajLieskovsky** [helix-editor/helix#11263] [Q&A]
Hi, I want to use latexdiff which I'd like to call the `\ShellEscape{}` command. For the building my latex projects I use tectonic from the texlab LS. When I try to build the log reads:
```
**
(diff.tex
LaTeX2e <2021-11-15> patch level 1
L3 programming layer <2022-02-24> (shellesc.sty
Package: shellesc 2019/11/08 v1.0c unified shell escape interface for LaTeX
Package shellesc Info: Unrestricted shell escape enabled on input line 75.
)
! ! LaTeX Error: File `main.tex' not found..
\@missingfileerror ...or: File `#1.#2' not found.}
                                                  
l.3 \input{main}
                
No pages of output.
```
where main.tex is the file that should be produces by latexdiff (`\ShellEscape{latexdiff main_og.tex main_rev.tex > main.tex}`).
My texlab config otherwise (without shell-escape) works well
```
[language-server.texlab.config.texlab.chktex]
onOpenAndSave = true
onEdit = true

[language-server.texlab.config.texlab.forwardSearch]
executable = "zathura"
args = [ "--synctex-forward", "%l:%c:%f", "%p" ]

[language-server.texlab.config.texlab.build]
auxDirectory = "build"
logDirectory = "build"
pdfDirectory = "build"

forwardSearchAfter = true
onSave = true

executable = "tectonic"
args = [
 "-X",
 "compile",
 "%f",
 "--synctex",
 "--keep-logs",
 "--keep-intermediates",
 "--outdir=build",
 "-Z",
 "shell-escape-cwd=$(pwd)",
 ]
```
and calling tectonic from the command line produces the document as intended `tectonic -X compile diff.tex -Z shell-escape-cwd=$(pwd)`. Any ideas what could be the issue?

**Comment by kirawi** [comment]
Substitutions are not correctly implemented so everything is passed literally to the tool. This needs https://github.com/helix-editor/helix/pull/11164

**Comment by pascalkuthe** [comment]
The %f substitution is handeled by this LSP. It's not expected that the editor does this.


I would raise this question to the texlab people. I have been using texlab with pdflatex and shell escape without issue.

The kne thing that may not work is `$(pwd)`. I am not sure how twxlab executes commands. You may need to pass this trough a shell first. (Using bash as the command and "-C" "tectonic").

I am not sure if specifying the working directory liek this is necessary tough. Doesn't tectonic have a normal -shell-escape flag. Usually child processes have the same working directory as their parent so it should use the right directory on its own
